src/main.py

```diff
@@ -20,56 +51,69 @@
-        if test_result != Result.SUCCESS:
-            test_group_failures += 1
-            # Mark that there is an error in this pod
-            test_group.slack_message.should_send = True
-        diagnostic_text += "[{}] - {}\n".format(test.name, test_result.name)
-    slack_message_text += "> ```\n{}```\n".format(diagnostic_text)
-
-    if test_group_failures:
-        passed_tests = num_test_group_tests - test_group_failures
-        slack_message_text += "\t*Summary: `{}/{}` tests passed!* ".format(passed_tests, num_test_group_tests)
-        # Tag appropriate users
-        user_ids = environ["ADMIN_SLACK_USER_IDS"]
-        if test_group.application in application_user_id_mapping:
-            user_ids += ", " + application_user_id_mapping[test_group.application]
-        slack_message_text += "cc {}!".format(user_ids)
-    else:
-        slack_message_text = "*`{0}/{0}` tests passed :white_check_mark:*".format(num_test_group_tests)
-    test_group.slack_message.text = slack_message_text
-
-    # Always print output for debugging purposes
-    print(test_group.slack_message.text)
+
+for i in range(len(test_groups)):
+    if test_config[i]!="0":  # Only test enabled or previously failing test groups
+        test_group=test_groups[i]
+        slack_message_text = "\tRunning tests for {}...\n".format(test_group.name)
+        test_group_failures = 0
+        num_test_group_tests = len(test_group.tests)
+
+        diagnostic_text = ""
+        for test in test_group.tests:
+            test_status = SUCCESS_STATUS  # default to printing success
+            test_result = test.get_result()
+            if test_result != Result.SUCCESS:
+                test_group_failures += 1
+                # Mark that there is an error in this pod
+                test_group.slack_message.should_send = True
+            diagnostic_text += "[{}] - {}\n".format(test.name, test_result.name)
+        slack_message_text += "> ```\n{}```\n".format(diagnostic_text)
+
+        if test_group_failures:
+            test_config[i]="2" # The test group has failed
+            passed_tests = num_test_group_tests - test_group_failures
+            slack_message_text += "\t*Summary: `{}/{}` tests passed!* ".format(passed_tests, num_test_group_tests)
+            # Tag appropriate users
+            user_ids = environ["ADMIN_SLACK_USER_IDS"]
+            if test_group.application in application_user_id_mapping:
+                user_ids += ", " + application_user_id_mapping[test_group.application]
+            slack_message_text += "cc {}!".format(user_ids)
+        else:
+            test_config[i]="1" # The test group has passed
+            slack_message_text = "*`{0}/{0}` tests passed :white_check_mark:*".format(num_test_group_tests)
+        test_group.slack_message.text = slack_message_text
+        # Always print output for debugging purposes
+        print(test_group.slack_message.text)
+
+print(test_config)
+
+f = open("./test_config.txt", "w")
```

Comment:
where would we be writing this?

Reply:
The code updates the config with updated tests

Reply:
replaced with a json